src/main.ts

```diff
@@ -0,0 +1,67 @@
+import { NestFactory } from '@nestjs/core'
+import { AppModule } from './app.module'
+import { ConfigService } from '@nestjs/config'
+import { Logger, VersioningType } from '@nestjs/common'
+import { DocumentBuilder, SwaggerModule } from '@nestjs/swagger'
+import { getLogLevels } from './config/logger.config'
+import * as fs from 'fs'
+import { WsAdapter } from './config/ws.adapter'
+
+/**
+ * Bootstraps the NestJS application, setting up configurations, middleware, and documentation.
+ *
+ * @returns {Promise<void>} - A promise that resolves when the application has started successfully.
+ */
+async function bootstrap(): Promise<void> {
+  // Retrieve log levels based on environment configuration
+  const logLevels = getLogLevels()
+
+  // Create the NestJS application with custom logger levels
+  const app = await NestFactory.create(AppModule, {
+    logger: logLevels,
+  })
+
+  const configService = app.get(ConfigService)
+  const logger = new Logger(bootstrap.name)
+
+  // Set up WebSocket adapter
+  app.useWebSocketAdapter(new WsAdapter(app) as any)
```

Comment:
No! this be used to change ws adapter in nestjs https://docs.nestjs.com/websockets/adapter, It will no longer be used due to the rpc-websocket library change.